src/cocotb/share/lib/vpi/VpiCbHdl.cpp

```diff
@@ -239,14 +239,14 @@ int VpiSignalObjHdl::initialise(const std::string &name,
     } else {
         m_num_elems = vpi_get(vpiSize, GpiObjHdl::get_handle<vpiHandle>());
 
-        if (GpiObjHdl::get_type() == GPI_STRING) {
+        if (vpiConstant == type || vpiParameter == type ||
+            vpiStringVar == type) {
```

Comment:
No need to replace what's there.
```suggestion
        if (GpiObjHdl::get_type() == GPI_STRING || type == vpiConstant ||
            type == vpiParameter) {
```